### PR TITLE
enhancement(prometheus_scrape source): comply with component spec

### LINKED ISSUE #10075: Instrument the `prometheus_scrape` source with event processing metrics
Per #9326

### PR DESCRIPTION
**Pull request by jdrouet**
- rename and update `PrometheusEventReceived`
- update errors tags
- add `StreamClosedError`
- add `BytesReceived`

closes #10075

Signed-off-by: Jérémie Drouet <jeremie.drouet@datadoghq.com>

### CODE REVIEW

File: src/sources/prometheus/scrape.rs [hunk 1]
```diff
@@ -255,7 +259,7 @@ fn prometheus(
 
                                 match parser::parse_text(&body) {
                                     Ok(events) => {
-                                        emit!(&PrometheusEventReceived {
+                                        emit!(&PrometheusEventsReceived {
                                             byte_size,
```
Comment (jszwedko) [inline src/sources/prometheus/scrape.rs:263]: I don't think this `byte_size` is right. It is the same as what is being used for `component_received_bytes_total`. Instead, it should be summing up `ByteSizeOf::size_of` the events.

**Review by jszwedko — CHANGES_REQUESTED** [review]
One question about the accounting of `component_received_event_bytes_total`. Otherwise looks good!

**Review by 001wwang — COMMENTED** [review]
Some recommended edits for consistency.

File: src/internal_events/prometheus.rs [hunk 2]
```diff
@@ -96,19 +114,22 @@ impl InternalEvent for PrometheusHttpResponseError {
...
         );
     }
 
     fn emit_metrics(&self) {
-        counter!("http_error_response_total", 1);
         counter!(
             "component_errors_total", 1,
             "code" => self.code.to_string(),
             "url" => self.url.to_string(),
+            "error" => self.code.canonical_reason().unwrap_or("unknown status code"),
```
[the remark this answers is not in the document]
Comment (jdrouet) [reply src/internal_events/prometheus.rs:127]: not, a mistake

File: src/internal_events/prometheus.rs [hunk 3]
```diff
@@ -171,18 +198,20 @@ impl InternalEvent for PrometheusNoNameError {
         error!(
             message = "Could not decode timeseries.",
             error = "Decoded timeseries is missing the __name__ field.",
+            error_type = "parse_failed",
             stage = "processing",
-            internal_log_rate_secs = 5
```
Comment (jszwedko) [inline src/internal_events/prometheus.rs:175]: Did you mean to drop the rate limit here?

**Review by jszwedko — APPROVED** [review]
Looks good! I think we should re-add the dropped rate limit though.

File: src/internal_events/prometheus.rs (hunk 3 above)
Comment (jdrouet) [reply src/internal_events/prometheus.rs:175]: I forgot them, just added them all 😉

**Pull request merged**